### ISSUE TITLE
wrong assertions to check actual stats with reference stats

### ISSUE THREAD

**Issue by soxofaan** [eu-cdse/openeo-benchmarks#4]
https://github.com/eu-cdse/openeo-benchmarks/blob/673122734a0516b414853faa1f4e7157f65d8b65/tests/test_regression.py#L57-L61

This `zip` patterns does not what you intend: `zip` on dicts just works with the keys of the dicts, not the values, so you are comparing the keys of `output_dict` and `groundtruth_dict`

Instead I think it's better to use this patterns:

```python
assert output_dict == {
    k: pytest.approx(v, rel=TOLERANCE)
    for k,v in groundtruth_dict.items()
}
```

note:
- there is just one `assert` but comparing dictionaries, advantage is that you get a lot more context (all stat values) when there is a failure, instead of just the first mismatch
- with `pytest.approx()` it's better to be explicit if you are talking about relative tolerance or absolute tolerance

**Comment by soxofaan** [comment]
reopening as #5 is not closed yet

**Comment by HansVRP** [comment]
Introduced the following function to also assert that the keys are equal and the input are effectively dictionaries

```
def assert_dicts_approx_equal(dict1: dict, dict2: dict, tolerance: float = 0.01) -> None:
    """
    Compares two dictionaries for approximate equality.

    Args:
        dict1 (dict): The first dictionary to compare.
        dict2 (dict): The second dictionary to compare.
        tolerance (float, optional): The relative tolerance for comparing values. Defaults to 0.01.

    Raises:
        AssertionError: If the dictionaries are not approximately equal within the specified tolerance.
    """

    
    assert set(dict1.keys()) == set(dict2.keys()), "The keys in both dictionaries must match."
    
    for key in dict1.keys():
        assert pytest.approx(dict1[key], rel=tolerance) == dict2[key], "Values for key '{key}' are not approximately equal."
        
        
```

**Comment by soxofaan** [comment]
fixed with #5
